### PR TITLE
Implement synchronoous backfill in DetectionJobScheduler

### PR DESCRIPTION
**Pull request by cyy0714**
* function return after backfill is done

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java [hunk 1]
```diff
@@ -366,6 +370,44 @@ public Long runBackfill(long functionId, DateTime backfillStartTime, DateTime ba
     }
     return jobId;
   }
+  /**
+   * Different from asynchronous backfill in runBackfill, it will return after the backfill is done.
+   * @param functionId
+   * @param backfillStartTime
+   * @param backfillEndTime
+   * @param force
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java]: Please complete the comment for this variable.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java [hunk 2]
```diff
@@ -366,6 +370,44 @@ public Long runBackfill(long functionId, DateTime backfillStartTime, DateTime ba
...
+        }
+        retryCounter++;
+        LOG.warn("The backfill of anomaly function {} (task id: {}) failed. Retry count: {}", functionId, jobExecutionId, retryCounter);
+        jobExecutionId = runBackfill(functionId, backfillStartTime, backfillEndTime, force);
+      }
+      try {
+        TimeUnit.SECONDS.sleep(SYNC_SLEEP_SECONDS);
+      }
+      catch (InterruptedException e) {
+        LOG.warn("The backfill of anomaly function {} (task id: {}) is awakened.", functionId, jobExecutionId);
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java]: The thread that monitor the backfill job is awaken, not the backfill process. You might want to refine this message.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java [hunk 3]
```diff
@@ -367,7 +372,68 @@ public Long runBackfill(long functionId, DateTime backfillStartTime, DateTime ba
...
+              jobDTO.setStatus(JobConstants.JobStatus.FAILED);
+              jobDAO.save(jobDTO);
+            }
 
+            return;
+          }
+          hasFailedTask = true;
+          LOG.warn("The backfill of anomaly function {} (task id: {}) failed. Retry count: {}", functionId,
+              jobExecutionId, retryCounter);
+          jobExecutionId = runBackfill(functionId, backfillStartTime, backfillEndTime, force);
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java]: Retry should be done only once per job, and not for every failed task. Retrying the job in effect retry each task

Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java]: Updated. Thanks

**Pull request closed without merging**